### ISSUE TITLE
checkpoints not getting saved

### ISSUE THREAD

**Issue by AASHISHAG** [bedapudi6788/txt2txt#6]
hello, thank you for open-sourcing the code.

when i run 

python3 txt2txt.py

the checkpoints are not getting saved. could you please guide.

your code seems correct, not sure where is the error

**Comment by bedapudi6788** [comment]
Please post the entire output.

**Comment by AASHISHAG** [comment]
Please refer it here: https://drive.google.com/open?id=1KsV-nFGvV5NzsZEt0VEVHNQr-nooN3xx

**Comment by bedapudi6788** [comment]
use monitor=val_accuracy in ModelCheckpoint

**Comment by AASHISHAG** [comment]
unfortunately, still the same issue: 
i can't see any checkpoint file... only params file is created. please note im trying to just run the master code .. i did't do any changes

please guide
https://drive.google.com/open?id=1gKtxo9hel8BEU99H9f2jBs5eiUTNNXEB

**Comment by bedapudi6788** [comment]
Please copt paste the erroe text here.

**Comment by AASHISHAG** [comment]
```
6/6 [==============================] - 0s 26ms/step - loss: 4.3865e-05 - acc: 1.0000 - val_loss: 4.3448e-05 - val_acc: 1.0000
/usr/local/lib/python3.5/dist-packages/keras/callbacks.py:434: RuntimeWarning: Can save best model only with val_accuracy available, skipping.
  'skipping.' % (self.monitor), RuntimeWarning)
/usr/local/lib/python3.5/dist-packages/keras/engine/network.py:877: UserWarning: Layer lstm_2 was passed non-serializable keyword arguments: {'initial_state': [<tf.Tensor 'concatenate_1/concat:0' shape=(?, 256) dtype=float32>, <tf.Tensor 'concatenate_2/concat:0' shape=(?, 256) dtype=float32>]}. They will not be included in the serialized model (and thus will be missing at deserialization time).
  '. They will not be included '
```

**Comment by bedapudi6788** [comment]
You have to check what the name of the accuracy metric is. This is not a
txt2txt issue per se. For the keras version you are using, just check
stackoverflow for how to save keras model based on accuracy metric.

**Comment by AASHISHAG** [comment]
@bedapudi6788 : Last question. I want to train txt2txt on German data for punctuation correction. So, I provide input and output in LIST format to txt2txt.py, will it work?

or Will I need to go any other changes?

**Comment by AASHISHAG** [comment]
@bedapudi6788 : I passed input and output as a list to txt2txt for punctuation correction and when I used deepcorrect to test the result, I got following result.

I passed the training data itself. But output is consistently 3 words. Please guide.

>>> corrector.correct("in keiner dieser debatten wurden diese grundsätze irgendwie bestritten oder angezweifelt")
[[{'sequence': 'in keiner', 'prob': 0.26473349747315716}]]
>>> corrector.correct("es geht mir in diesem kontext darum spielraum zu schaffen für zwei dinge")
[[{'sequence': 'es geht m', 'prob': 0.1473360733543895}]]
